File: employee-app/src/app/services/employee.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

export interface Employee {
  id: number;
  department: string;
  fullName: string;
  birthDate: string;
  hireDate: string;
  salary: number;
}

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private apiUrl = 'https://localhost:7226/api/employees';

  constructor(private http: HttpClient) { }

  getEmployees(): Observable<Employee[]> {
    return this.http.get<Employee[]>(this.apiUrl);
  }

  createEmployee(employee: Employee): Observable<any> {
  return this.http.post(this.apiUrl, employee);
}

updateEmployee(employee: Employee): Observable<any> {
  return this.http.put(`${this.apiUrl}/${employee.id}`, employee);
}

deleteEmployee(id: number): Observable<any> {
  return this.http.delete(`${this.apiUrl}/${id}`);
}

}

File: EmployeeAPI/EmployeeAPI/Models/Employee.cs
﻿namespace EmployeeApi.Models
{
    public class Employee
    {
        public int Id { get; set; }
        public string Department { get; set; }
        public string FullName { get; set; }
        public DateTime BirthDate { get; set; }
        public DateTime HireDate { get; set; }
        public decimal Salary { get; set; }
    }
}
